src/web/routes/application/middleware/handle-path-request/handle-path-request.js

```diff
@@ -1,20 +1,21 @@
-const { CHECK_URL, TERMS_AND_CONDITIONS_URL, CONFIRM_URL } = require('../common/constants')
-const { isGuidancePageUrl } = require('../../guidance/predicates')
-const { stateMachine, actions, states } = require('../common/state-machine')
+const { CONFIRM_URL } = require('../../common/constants')
+const { stateMachine, actions, states } = require('../../common/state-machine')
+const { isPathInSequence, stepNotNavigable } = require('./predicates')
+const { getPathsInSequence } = require('./selectors')
 
 const { IS_PATH_ALLOWED, GET_NEXT_ALLOWED_PATH } = actions
 
-const getPathsInSequence = (steps) => [...steps.map(step => step.path), CHECK_URL, TERMS_AND_CONDITIONS_URL, CONFIRM_URL]
-
-const stepNotNavigable = (step, req) => step && typeof step.isNavigable === 'function' && !step.isNavigable(req.session)
-
 const middleware = (config, pathsInSequence, step) => (req, res, next) => {
   // Destroy the session on navigating away from CONFIRM_URL
   if (stateMachine.getState(req) === states.COMPLETED && req.path !== CONFIRM_URL) {
     req.session.destroy()
     res.clearCookie('lang')
-    const redirectPath = isGuidancePageUrl(req.path) ? req.path : config.environment.OVERVIEW_URL
-    return res.redirect(redirectPath)
+
+    if (isPathInSequence(req.path, pathsInSequence)) {
```

Comment:
This doesn't read quite right, I know that's what the function does but I think it needs to be more descriptive of the scenario its dealing with so might need to be something like `isPathInApplicationFlow`. Let's discuss when you get in.

Reply:
Renamed to `isPathInApplicationFlow`